storage/remote/client.go

```diff
@@ -154,6 +161,11 @@ type ReadClient interface {
 	Read(ctx context.Context, query *prompb.Query, sortSeries bool) (storage.SeriesSet, error)
 }
 
+// SetUserAgent allows the User-Agent header value to be over-ridden.
+func SetUserAgent(userAgent string) {
```

Comment:
What does this change?  It could already be overridden simply by assigning to it.

Reply:
Sure, it could always have been overridden by assigning to it but having a dedicated function makes it more explicit (given that the original issue could have been avoided by simply setting the `UserAgent` value anyway). Also by doing so we can preserve the original version (in an unmodifiable value/header) to allow client versions to be tracked in the future.